resources/qml/delegates/PlayableMediaMessage.qml

```diff
@@ -5,22 +5,64 @@ import QtMultimedia 5.6
 
 import im.nheko 1.0
 
-Column { 
+ColumnLayout { 
+    // duration = milliseconds
+	function durationToString(duration) {
+		function maybeZeroPrepend(time) {
+			return (time < 10) ? "0" + time.toString() :
+				time.toString()
+		}
+		var totalSeconds = Math.floor(duration / 1000)
+		var seconds = totalSeconds % 60
+		var minutes = (Math.floor(totalSeconds / 60)) % 60
+		var hours = (Math.floor(totalSeconds / (60 * 24))) % 24
+		// Always show minutes and don't prepend zero into the leftmost element
+		var ss = maybeZeroPrepend(seconds)
+		var mm = (hours > 0) ? maybeZeroPrepend(minutes) : minutes.toString()
+		var hh = hours.toString()
+
+		if (hours < 1)
+			return mm + ":" + ss
+		return hh + ":" + mm + ":" + ss
+	}
+	
     id: content
-    width: parent.width - 24
-    anchors.centerIn: parent
+    Layout.maximumWidth: parent ? parent.width: undefined
+
+    MediaPlayer {
+        id: media
+        onError: console.log(errorString)
+        volume: volumeSlider.desiredVolume
+    }
+
+    Connections {
+        property bool mediaCached: false
 
+        id: mediaCachedObserver
+        target: TimelineManager.timeline
+        onMediaCached: {
+            if (mxcUrl == model.data.url) {
+                mediaCached = true
+                media.source = "file://" + cacheUrl
+                console.log("media loaded: " + mxcUrl + " at " + cacheUrl)
+            }
+            console.log("media cached: " + mxcUrl + " at " + cacheUrl)
+        }
+    }
+      
     Rectangle {
         id: videoContainer
         visible: model.data.type == MtxEvent.VideoMessage
-        property double tempWidth: Math.min(parent ? parent.width : undefined, model.data.width < 1 ? 400 : model.data.width)
+        //property double tempWidth: Math.min(parent ? parent.width : undefined, model.data.width < 1 ? 400 : /////model.data.width)
+        property double tempWidth: (model.data.width < 1) ? 400 : model.data.width
```

Comment:
I think this is breaking your sizing. You should be using Math.min, so that it is using the minimum of the parent.width or the video dimensions.